app/services/katello/candlepin/package_profile_uploader.rb

```diff
@@ -0,0 +1,31 @@
+module Katello
+  module Candlepin
+    class PackageProfileUploader
+      def self.upload(profile_string:, host:)
+        host_id = host.id
+        profile = JSON.parse(profile_string)
+        Rails.logger.info "SIZE: #{profile_string.size.to_s(:human_size)}"
+        #free the huge string from the memory
+        profile_string = 'TRIMMED'.freeze
+        import_package_profile_for_host(host_id, profile)
+        ::Katello::Host::ContentFacet.trigger_applicability_generation(host_id)
+      end
+
+      def self.import_package_profile_for_host(host_id, profile)
+        host = ::Host.find_by(:id => host_id)
```

Comment:
The `upload` method already had the host object, and this code takes the ID and looks it up again

Reply:
This is because I was trying to preserve the following logic and not alter it (my mindset was that this is a refactor, so only make changes that are semantically necessary due to the code moves.)

```ruby
if host.nil?
          Rails.logger.warn("Host with ID %s not found; continuing" % host_id)
        elsif host.content_facet.nil? || host.content_facet.uuid.nil?
          Rails.logger.warn("Host with ID %s has no content facet; continuing" % host_id)
        else
...
```

however, I'm failing to see how `host` would ever be nil anyway. Should those warning logs be moved to the controller, maybe? Open to suggestions..